src/app/home/home.page.ts

```diff
@@ -74,102 +76,101 @@ export class HomePage implements OnInit {
       }
     });
   }
-  createLetters() {
+
+  private createLetters() {
     this.letters = Object.keys(this.groupedCountries).sort();
   }
 
- 
+  private updatedLoadingState(item: Country): void {
+    this.isLoaded = true;
+    this.isLoading = false;
+    this.loadedItem = item;
+    this.loadingItem = null;
+  }
+
+  private addToFavorites(item: Country): void {
+    if (!this.favourites.includes(item))
+      this.favourites.push(item);
+  }
+
+  private removeFromFavorites(item: Country): void {
```

Comment:
use   Array.prototype.filter() for   this.favourites array to include only those items (c) that are not equal (!==) to item. Essentially, it removes item from this.favourites